examples/azure_ml_orchestrator_flavor/orchestrator/azure_ml_pipelines_orchestrator.py

```diff
@@ -73,356 +69,337 @@ def settings_class(self) -> Optional[Type["BaseSettings"]]:
-                        f"the AzureMLPipelines pipeline step.\n"
-                        f"Please ensure that you always use non-local "
-                        f"stack components with a remote AzureMLPipelines orchestrator, "
-                        f"otherwise you may run into pipeline execution "
-                        f"problems. You should use a flavor of "
-                        f"{stack_comp.type.value} other than "
-                        f"'{stack_comp.flavor}'.\n"
-                        + silence_local_validations_msg
-                    )
-
-                # if the orchestrator is remote, the container registry must
-                # also be remote.
-                if container_registry.config.is_local:
-                    return False, (
-                        f"{msg}the AzureMLPipelines orchestrator is configured to run "
-                        f"pipelines in a remote Kubernetes cluster, but the "
-                        f"'{container_registry.name}' container registry URI "
-                        f"'{container_registry.config.uri}' "
-                        f"points to a local container registry. Please ensure "
-                        f"that you always use non-local stack components with "
-                        f"a remote AzureMLPipelines orchestrator, otherwise you will "
-                        f"run into problems. You should use a flavor of "
-                        f"container registry other than "
-                        f"'{container_registry.flavor}'.\n"
-                        + silence_local_validations_msg
-                    )
-
-            return True, ""
-
-        return StackValidator(
-            required_components={
-                StackComponentType.CONTAINER_REGISTRY,
-                StackComponentType.IMAGE_BUILDER,
-            },
-            custom_validation_function=_validate,
-        )
-
-    def prepare_or_run_pipeline(
-        self,
-        deployment: "PipelineDeploymentResponseModel",
-        stack: "Stack",
-        environment: Dict[str, str],
-    ) -> Any:
-        """Runs the pipeline on AzureMLPipelines.
-
-        This function first compiles the ZenML pipeline into a AzureMLPipelines yaml
-        and then applies this configuration to run the pipeline.
-
-        Args:
-            deployment: The pipeline deployment to prepare or run.
-            stack: The stack the pipeline will run on.
-            environment: Environment variables to set in the orchestration
-                environment.
+            The orchestrator run id.
 
         Raises:
-            RuntimeError: If you try to run the pipelines in a notebook environment.
+            RuntimeError: If the run id cannot be read from the environment.
         """
-        # First check whether the code running in a notebook
-        if Environment.in_notebook():
+        try:
+            return os.environ[ENV_ZENML_AZUREML_RUN_ID]
```

Comment:
For your pipeline to actually run on AzureML, you'd need to make sure this environment variable is set.

Best would be to use an environment variable that already exists and is getting set by AzureML itself, e.g. if they have a unique ID for the pipeline run that is happening on their side. The value that you return from this method needs to be
- different for each pipeline run
- same for all steps of a single pipeline run

Maybe it's also possible to use the `Run` class (https://learn.microsoft.com/en-us/python/api/azureml-core/azureml.core.run%28class%29?view=azure-ml-py) of the AzureML SDK to get the current run. Note that the `get_run_id()` method will only be called inside the Docker image running on AzureML, not on your local machine.